transport/src/main/java/io/netty/channel/pool/ChannelPool.java

```diff
@@ -58,4 +58,11 @@
 
     @Override
     void close();
+
+    /**
+     * Closes the pool in an async manner.
+     *
+     * @return Future which represents completion of the close task
+     */
+    Future<Void> closeAsync();
```

Comment:
Unfortunately we can not do this as this would be an API breaking change.

Reply:
Pushed new revision without the interface change